### PR TITLE
Feature/login with helium

### PR DESCRIPTION
**Pull request by matthewcarlreetz**
This PR must be used in tandem with
https://github.com/helium/hotspot-app/pull/1133

### CODE REVIEW

File: .expo/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+> Why do I have a folder named ".expo" in my project?
```
Comment (tyler-whitman) [inline .expo/README.md:1]: I think we can add the `.expo` folder to git ignore

File: android/app/build.gradle [hunk 2]
```diff
@@ -9,7 +9,6 @@ buildscript {
 
 apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'
 apply plugin: "com.android.application"
-apply from: '../../node_modules/react-native-unimodules/gradle.groovy'
```
Comment (tyler-whitman) [inline android/app/build.gradle:12]: We had a line:

```
apply from: new File(["node", "--print", "require.resolve('react-native/package.json')"].execute().text.trim(), "../react.gradle")
```
in the hotspot app. I saw that in the migration guild, was it not needed?

File: android/build.gradle [hunk 3]
```diff
@@ -5,33 +5,36 @@ buildscript {
...
         minSdkVersion = 21
         compileSdkVersion = 30
-        targetSdkVersion = 29
-        ndkVersion = "20.1.5948944"
+        targetSdkVersion = 30
+        ndkVersion = "21.4.7075529"
     }
     repositories {
         google()
         jcenter()
```
Comment (tyler-whitman) [inline android/build.gradle:13]: I think we should remove jcenter it has been deprecated

File: android/build.gradle [hunk 4]
```diff
@@ -5,33 +5,36 @@ buildscript {
...
         }
         maven {
             // Android JSC is installed from npm
-            url("$rootDir/../node_modules/jsc-android/dist")
+            url(new File(["node", "--print", "require.resolve('jsc-android/package.json')"].execute(null, rootDir).text.trim(), "../dist"))
         }
 
         google()
+        mavenCentral()
         jcenter()
```
Comment (tyler-whitman) [inline android/build.gradle:38]: Can remove it here too

File: src/App.tsx [hunk 5]
```diff
@@ -96,13 +95,13 @@ const App = () => {
 
   // hide splash screen
   useAsync(async () => {
-    const loggedOut = isRestored && !isBackedUp
-    const loggedInAndLoaded = isRestored && isBackedUp
+    const loggedOut = isRestored
+    const loggedInAndLoaded = isRestored
 
     if (loggedOut || loggedInAndLoaded) {
```
Comment (tyler-whitman) [inline src/App.tsx:101]: Maybe we just change this to `if (isRestored)`

File: src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx [hunk 6]
```diff
@@ -1,183 +1,125 @@
...
+        owner: ownerAddress,
+        // currentLocation: '', // If reasserting location, put previous location here
+        makerAddress: onboardingRecord.maker.address,
+        locationNonceLimit: onboardingRecord.maker.locationNonceLimit || 0,
+      })
+      updateParams.assertLocationTxn = assertLocationTxn.toString()
     }
+
+    const url = WalletLink.createUpdateHotspotUrl(updateParams)
+    if (!url || !Linking.canOpenURL(url)) return
```
Comment (tyler-whitman) [inline src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx:111]: Should we show an alert or something?

**Review by tyler-whitman — CHANGES_REQUESTED** [review]

File: android/app/build.gradle (hunk 2 above)
Comment (matthewcarlreetz) [reply android/app/build.gradle:12]: hmmm. I just missed this, but it worked fine without 🤷‍♂️

File: src/App.tsx (hunk 5 above)
Comment (matthewcarlreetz) [reply src/App.tsx:101]: ha, yeah.

File: android/build.gradle (hunk 3 above)
Comment (matthewcarlreetz) [reply android/build.gradle]: The guide keeps it there, I think I'll leave it, but move it below mavenCentral. According to jcenter they are going to keep it available as read only.

File: src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx (hunk 6 above)
Comment (matthewcarlreetz) [reply src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx:111]: We could, but I guess I'd rather leave how it's handled up to the makers.

**Pull request merged**